kolibri_instant_schools_plugin/auth/api.py

```diff
@@ -197,20 +198,21 @@ def create(self, request):
 
         return Response(username, status=status.HTTP_201_CREATED)
 
-    def list(self, request):
+    @action(methods=["POST"], detail=False)
+    def profiles(self, request):
         """
         Get a list of profiles associated with a phone number (authenticated by password).
 
         Usage:
 
-            GET /user/api/phoneaccountprofile/?phone=<phone>&password=<password>
+            POST {"phone": "<phone>", "password": "<password>"} to /user/api/phoneaccountprofile/
                 If successful, returns status 200 with a list of dicts with `full_name` and `username`.
                 If no accounts are found, returns status 404.
                 If password fails, returns status 401.
         """
         # extract the phone and password from the query params
-        phone = normalize_phone_number(request.query_params.get('phone', ''))
-        password = request.query_params.get('password', '')
+        phone = normalize_phone_number(request.data['phone'])
```

Comment:
This looks right to me.

The only thing you might want to add is some validation that `phone` and `password` are defined, otherwise this API endpoint will give a 500. Best to check for existence of both and raise a 400 if not.